libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/offboard/MapboxOffboardRouter.kt

```diff
@@ -70,22 +71,48 @@ class MapboxOffboardRouter(
                     response: Response<DirectionsResponse>
                 ) {
                     directionRequests.remove(requestId)
+                    val url = call.request().url.toUrl()
                     val routes = response.body()?.routes()
+                    val code = response.code()
                     when {
-                        call.isCanceled -> callback.onCanceled()
-                        response.isSuccessful && !routes.isNullOrEmpty() -> {
-                            callback.onResponse(routes)
+                        call.isCanceled -> callback.onCanceled(routeOptions)
+                        response.isSuccessful -> {
+                            if (!routes.isNullOrEmpty()) {
+                                callback.onRoutesReady(routes)
+                            } else {
+                                callback.onFailure(
+                                    listOf(
+                                        RouterFailure(url, ROUTES_LIST_EMPTY, code)
+                                    ),
+                                    routeOptions
+                                )
+                            }
```

Comment:
Didn't we agree that this is not an option? cc @danpat @mskurydin @vanmaxim

Reply:
It should never occur - if it does, it's a server-side bug.